scalafix-core/shared/src/main/scala/scalafix/internal/config/DisableConfig.scala

```diff
@@ -2,18 +2,85 @@ package scalafix
 package internal.config
 
 import metaconfig._
+import metaconfig.annotation.{Description, ExampleValue}
+import metaconfig.generic.Surface
 import org.langmeta.Symbol
 
 import scalafix.internal.util.SymbolOps
-import metaconfig.annotation.{Description, ExampleValue}
-import metaconfig.generic.Surface
+
+case class DisabledSymbol(
+    @Description("Symbol to ban.")
+    symbol: Option[Symbol.Global],
+    @Description("Custom message.")
+    message: Option[String],
+    @Description("Custom id for error messages.")
+    id: Option[String],
+    @Description(
+      "Regex to ban instead of one symbol. " +
+        "Supports exclude and include regexes." +
+        "Symbol option is forbidden when regex is specified.")
+    @ExampleValue("""
+                    |{
+                    |  include = "java.io.*"
```

Comment:
Is this example copy-pasteable? `include` is a HOCON keyword.

It might also be good to give an example with lists
```
includes = [
  "java.io.*"
  "scala.io.*"
]
excludes = [
  scala.io.Foo
  // ...
]
```

Reply:
would this allow the use of `Foo` the type and the `Foo` constructor whilst banning use of methods on `Foo`?

Reply:
You can probably encode those rules with more advanced regexps. At the end of the day we have a bunch of "symbol occurences" where symbols are opaque string IDs following a well-specified format https://github.com/scalameta/scalameta/blob/master/semanticdb/semanticdb3/semanticdb3.md#scala-symbol

I think the current implementation uses "normalized" symbol matching, which blurs between type/term equality but IIRC we can make "exact" symbol matching opt-in so you can target the type explicitly with `Foo#` instead of `Foo.` (term)

Reply:
In SemanticDB, a `Symbol` is nothing more than a string ID. `SymbolInformation` holds metadata like types/modifiers/.

Every class/object/def/val/type parameter/.. has a globally unique symbol that should also be quite readable and ergonomic to write by hand.